components/match2/wikis/trackmania/match_group_input_custom.lua

```diff
@@ -0,0 +1,462 @@
+	for opponentIndex = 1, MAX_NUM_OPPONENTS do
+		-- read opponent
+		local opponent = args['opponent' .. opponentIndex]
+		if not Logic.isEmpty(opponent) then
+			CustomMatchGroupInput.processOpponent(opponent, args.date)
+
+			-- Retrieve icon for team
+			if opponent.type == Opponent.team then
+				opponent.icon, opponent.icondark = opponentFunctions.getTeamIcon(opponent.template)
+			end
+
+			-- apply status
+			if TypeUtil.isNumeric(opponent.score) then
+				opponent.status = STATUS_HAS_SCORE
+				isScoreSet = true
+			elseif Table.includes(ALLOWED_STATUSES, opponent.score) then
+				opponent.status = opponent.score
+				opponent.score = -1
+			end
+
+			--set Walkover from Opponent status
+			args.walkover = args.walkover or STATUS_TO_WALKOVER[opponent.status]
+
+			opponents[opponentIndex] = opponent
+
+			-- get players from vars for teams
+			if opponent.type == 'team' and not Logic.isEmpty(opponent.name) then
+				args = matchFunctions.getPlayers(args, opponentIndex, opponent.name)
+			end
+		end
+	end
+
+	--set resulttype to 'default' if walkover is set
+	if args.walkover then
+		args.resulttype = 'default'
+	end
+
+	local autofinished = String.isNotEmpty(args.autofinished) and args.autofinished or true
+	-- see if match should actually be finished if score is set
+	if isScoreSet and Logic.readBool(autofinished) and not Logic.readBool(args.finished) then
+		local currentUnixTime = os.time(os.date('!*t'))
+		local lang = mw.getContentLanguage()
+		local matchUnixTime = tonumber(lang:formatDate('U', args.date))
+		local threshold = args.dateexact and 30800 or 86400
+		if matchUnixTime + threshold < currentUnixTime then
+			args.finished = true
+		end
+	end
+
+	-- apply placements and winner if finshed
+	local winner = tostring(args.winner or '')
+	if Logic.readBool(args.finished) then
+		local placement = 1
+		local lastScore
+		local lastPlacement = 1
+		local lastStatus
+		-- luacheck: push ignore
+		for opponentIndex, opponent in Table.iter.spairs(opponents, CustomMatchGroupInput._placementSortFunction) do
+			if opponent.status ~= STATUS_HAS_SCORE and opponent.status ~= STATUS_DEFAULT_WIN and placement == 1 then
+				placement = 2
+			end
+			if placement == 1 then
+				args.winner = opponentIndex
+			end
+			if opponent.status == STATUS_HAS_SCORE and opponent.score == lastScore then
+				opponent.placement = lastPlacement
+			elseif opponent.status ~= STATUS_HAS_SCORE and opponent.status == lastStatus then
+				opponent.placement = lastPlacement
+			else
+				opponent.placement = placement
+			end
+			args['opponent' .. opponentIndex] = opponent
+			placement = placement + 1
+			lastScore = opponent.score
+			lastPlacement = opponent.placement
+			lastStatus = opponent.status
+		end
+	-- luacheck: pop
+	-- only apply arg changes otherwise
+	else
+		for opponentIndex, opponent in pairs(opponents) do
+			args['opponent' .. opponentIndex] = opponent
+		end
+	end
+	if
+		winner == RESULT_TYPE_DRAW or
+		winner == '0' or (
```

Comment:
imo const var

Reply:
ok, but we use this 0 two times at line 332 and 340 also, so should we make a const var that we convert into a String in this condition ? So we can use it at those two lines behind ?

Reply:
could just cast winner in `tonumber` in this line

Reply:
https://github.com/Liquipedia/Lua-Modules/pull/2273/commits/90cbac34c44f6d75430056c4372a07af40d91fd2